mubench.reviewsite/src/Controllers/RunsController.php

```diff
@@ -90,9 +90,14 @@ public function postRun(Request $request, Response $response, array $args)
         if (!$run) {
             return error_response($response,400, "empty: " . print_r($_POST, true));
         }
+
         $hits = $run->{'potential_hits'};
         $this->logger->info("received data for '" . $experimentId . "', '" . $detector_muid . "." . $project_muid . "." . $version_muid . "' with " . count($hits) . " potential hits.");
-        $this->addRun($experimentId, $detector_muid, $project_muid, $version_muid, $run);
+        try{
+            $this->addRun($experimentId, $detector_muid, $project_muid, $version_muid, $run);
+        }catch(\Exception $e){
+            return error_response($response, 400, $e->getMessage());
```

Comment:
I don't really like this handling via exceptions. This would also capture an SQLException, which would not be due to a bad request, but an internal server error... it drops the stacktrace, which could be important in this case. Either we use a dedicated exception type for bad requests and catch it here, or we find some alternative means for error handling without exceptions.  After all, it's not an exceptional case we're treating here, but part of the application logic.